Python/bytecodes.c

```diff
@@ -3973,7 +3968,12 @@ dummy_func(
             ERROR_IF(next_opcode < 0, error);
             next_instr = this_instr;
             if (_PyOpcode_Caches[next_opcode]) {
-                INCREMENT_ADAPTIVE_COUNTER(this_instr[1].cache);
+                if (next_opcode != POP_JUMP_IF_FALSE &&
+                    next_opcode != POP_JUMP_IF_TRUE &&
+                    next_opcode != POP_JUMP_IF_NOT_NONE &&
+                    next_opcode != POP_JUMP_IF_NONE) {
+                    INCREMENT_ADAPTIVE_COUNTER(next_instr[1].cache);
+                }
```

Comment:
@markshannon I had to add this because the naive increment would occasionally mess with the bitmap of recent branches used by the `POP_JUMP_IF` opcodes, and I had added an assert that complains if `backoff > 12`. I guess it was previously also messing with the bitmap, but nobody cared, and the effect is rarely significant. But this also makes me wonder if we don't need a different primitive than `INCREMENT_ADAPTIVE_COUNTER()`.

Reply:
Okay, I'll just do `value |= 1`.